airflow/plugins_manager.py

```diff
@@ -67,6 +67,8 @@ class AirflowPluginException(Exception):
 class AirflowPlugin:
     """Class used to define AirflowPlugin."""
     name: Optional[str] = None
+    plugin_source: Optional[str] = None
+    plugin_path: Optional[str] = None
```

Comment:
Thought: Not sure this should be part of the plugin interface. This is the class the plugin inherits from, and has so far been  necessarily the data structure(D) that Airflow owns to track discovered plugins.

Also, the name `plugin_path` is overloaded here; it is only a path for one type of plugin, so I'd use `plugin_location` perhaps?

Reply:
Coming back to this: if it is decided to keep this information on the AirflowPlugin subclasses, then it should be marked as 'internal', by using names that start with an underscore: `_plugin_source` and `_plugin_path`. 

Another thought: should this be a *single* attribute? It could be a data class along the lines of:

```
class BasePluginSource:
    source = None
    def __str__(self):
        raise NotImplementedError
    def __html__(self):
        raise NotImplementedError

class PluginsDirectorySource(BasePluginSource):
    source = "airflow_plugins_directory"
    def __init__(self, path):
        self.path = os.path.relpath(path, settings.PLUGINS_FOLDER)
    def __str__(self):
        return f"$PLUGINS_FOLDER/{self.path}"
    def __html__(self):
        return f"<em>$PLUGINS_FOLDER/</em>{self.path}"

class EntryPointSource(BasePluginSource):
    source = "entrypoint"
    def __init__(self, entrypoint):
        self.entrypoint = str(entrypoint)
        self.dist = str(entrypoint.dist)
    def __str__(self):
        return f"{self.entrypoint}: {self.dist}"
    def __html__(self):
        return f"<em>{self.entrypoint}:</em> {self.dist}"
```

The `__html__` method will be called by Jinja instead of `__str__`; the markup is just a sketch, feel free to provide better markup.

Reply:
Great suggestions. I have made changes related to this.